supernode/server/api/utils.go

```diff
@@ -52,26 +52,28 @@ func ParseJSONRequest(req io.Reader, target interface{}, validator ValidateFunc)
 	return nil
 }
 
-// EncodeResponse encodes response in json.
-// The response body is empty if the data is nil or empty value.
-func EncodeResponse(w http.ResponseWriter, code int, data interface{}) error {
+// SendResponse encodes response in json.
+//
+// TODO:
+//  Should the response body should be empty if the data is nil or empty
+//  string? Now it's incompatible with the client.
```

Comment:
I don't think we should. If the data is nil/empty, it basically means there is an error somehow, in that case, the `code` should reflect which error it is, and data should be used to describe what error it is.

Reply:
Ok, maybe we should provide a method to send header only.